uberfire-workbench/uberfire-workbench-processors/src/main/java/org/uberfire/annotations/processors/ClientEditorActivityGenerator.java

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.uberfire.annotations.processors;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.PackageElement;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic.Kind;
+
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import org.uberfire.annotations.processors.exceptions.GenerationException;
+import org.uberfire.annotations.processors.facades.ClientAPIModule;
+
+/**
+ * A source code generator for Activities
+ */
+public class ClientEditorActivityGenerator extends AbstractGenerator {
+
+    @Override
+    public StringBuffer generate(final String packageName,
+                                 final PackageElement packageElement,
+                                 final String className,
+                                 final Element element,
+                                 final ProcessingEnvironment processingEnvironment) throws GenerationException {
+
+        final Messager messager = processingEnvironment.getMessager();
+        messager.printMessage(Kind.NOTE,
+                              "Starting code generation for [" + className + "]");
+
+        //Extract required information
+        final TypeElement classElement = (TypeElement) element;
+
+        final String annotationName = ClientAPIModule.getWorkbenchClientEditorClass();
+        final boolean isDynamic = ClientAPIModule.getWbEditorIsDynamicValueOnClass(classElement);
+
+        String identifier = null;
+
+        for (final AnnotationMirror am : classElement.getAnnotationMirrors()) {
```

Comment:
imho this `for-cycle` could be extracted to a private method (`getIdentifier`) for better readability